ksql-engine/src/main/java/io/confluent/ksql/structured/SchemaKStream.java

```diff
@@ -184,12 +186,14 @@ public SchemaKStream into(
 
   public SchemaKStream<K> select(
       final List<SelectExpression> selectExpressions,
-      final QueryContext.Stacker contextStacker) {
+      final QueryContext.Stacker contextStacker,
+      final ProcessingLogContext processingLogContext) {
     final Selection selection = new Selection(
         selectExpressions,
-        ProcessingLoggerFactory.getLogger(
+        processingLogContext.getLoggerFactory().getLogger(
             QueryLoggerUtil.queryLoggerName(
-                contextStacker.push(Type.PROJECT.name()).getQueryContext())));
+                contextStacker.push(Type.PROJECT.name()).getQueryContext())),
+        processingLogContext);
```

Comment:
Is it standard practice to pass around both the context for an object and the object itself (in this case, `processingLogContext` and the logger itself)? Since each logger is only used with a single context, it feels like there should be a way to combine the two at the time the logger is created so they wouldn't need to be passed around separately. Not sure whether this is preferable though.

Reply:
I think standard practice is kind of orthogonal to whether it makes sense to do this here or not. I don't know if context is the right word for this object. It's an object that represents our interface to the processing logger module configured according to a given config. I don't see the issue with passing around both if the SelectValueMapper needs to use them both. I guess we could just pass the query context down instead of the logger and create that from SelectValueMapper. I'm kind of ambivalent about which is better.